androidHackathonApp/src/main/java/com/kcc/kmmhackathon/androidHackathonApp/view/SettingsFragment.kt

```diff
@@ -0,0 +1,33 @@
+package com.kcc.kmmhackathon.androidHackathonApp.view
+
+import androidx.lifecycle.ViewModelProvider
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.kcc.kmmhackathon.androidHackathonApp.R
+import com.kcc.kmmhackathon.androidHackathonApp.viewmodel.SettingsViewModel
+
+class SettingsFragment : Fragment() {
+
+    companion object {
+        fun newInstance() = SettingsFragment()
+    }
+
+    private lateinit var viewModel: SettingsViewModel
```

Comment:
I think you can do just `private val viewModel: SettingsViewModel by viewModels()`, although you might need another dependency for that

Reply:
Yep! This was the next task when moving over to the viewmodels :) although i've never done it this way, but read about it the other day

Reply:
This is what got auto-generated when creating a Fragment with ViewModel using the NavGraph editor

Reply:
Ah yeah some of the autogenerated stuff is a bit old sadly